File: backend/src/main/java/com/temka/app/entity/Submission.java
package com.temka.app.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;
import java.time.LocalDate;

@Entity
@Table(name = "submissions")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Submission {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User submittedBy;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private String country;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ProgramType type;

    private LocalDate deadline;

    @Column(length = 500)
    private String url;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private SubmissionStatus status = SubmissionStatus.PENDING;

    @Column(columnDefinition = "TEXT")
    private String adminComment;

    @Column(nullable = false, updatable = false)
    private Instant createdAt;

    private Instant reviewedAt;

    @PrePersist
    void prePersist() {
        createdAt = Instant.now();
    }
}

File: frontend/src/shared/types/submission.ts
import type { ProgramType } from "./program";

export type SubmissionStatus = "PENDING" | "APPROVED" | "REJECTED";

export type Submission = {
  id: number;
  userId: number;
  userName: string;
  title: string;
  description: string;
  country: string;
  type: ProgramType;
  deadline: string | null;
  url: string | null;
  status: SubmissionStatus;
  adminComment: string | null;
  createdAt: string;
  reviewedAt: string | null;
};

/**
 * Тело для POST /api/submissions и POST /api/admin/programs — на бэке это
 * SubmissionRequest и ProgramRequest, у них одинаковый набор полей.
 * deadline и url на бэке необязательны (@Future / @URL пропускают null).
 */
export type ProgramDraft = {
  title: string;
  description: string;
  country: string;
  type: ProgramType;
  deadline: string | null;
  url: string | null;
};

/** @deprecated Историческое имя, оставлено ради совместимости импортов. */
export type SuggestProgramRequest = ProgramDraft;
